pyramid_storage/gcloud.py

```diff
@@ -43,30 +46,49 @@ def from_settings(cls, settings, prefix):
             ('gcloud.credentials', False, None),
             ('gcloud.project', False, None),
             ('gcloud.bucket_name', True, None),
-            ('gcloud.acl', False, DEFAULT_FILE_ACL),
+            ('gcloud.acl', False, None),
             ('base_url', False, ''),
             ('extensions', False, 'default'),
             # Gcloud Connection options.
             ('gcloud.auto_create_bucket', False, False),
-            ('gcloud.auto_create_acl', False, DEFAULT_BUCKET_ACL),
+            ('gcloud.auto_create_acl', False, None),
             ('gcloud.cache_control', False, None),
+            ('gcloud.uniform_bucket_level_access', False, False),
         )
         kwargs = utils.read_settings(settings, options, prefix)
         kwargs = dict([(k.replace('gcloud.', ''), v) for k, v in kwargs.items()])
         return cls(**kwargs)
 
     def __init__(self, credentials, bucket_name, project=None, acl=None, base_url='',
                  extensions='default', auto_create_bucket=False,
-                 auto_create_acl=DEFAULT_BUCKET_ACL, cache_control=None):
+                 auto_create_acl=None, cache_control=None,
+                 uniform_bucket_level_access=False):
+        self.logger = logging.getLogger(__name__)
+
+        if (acl or auto_create_acl) and uniform_bucket_level_access:
+            raise ConfigurationError(
+                '"acl" and "auto_create_acl" \
+                        should remain unset \
+                        when "uniform_bucket_level_access" \
+                        is enabled!'
+            )
+
+        if not uniform_bucket_level_access:
+            self.acl = acl or DEFAULT_FILE_ACL
+            self.auto_create_acl = auto_create_acl or DEFAULT_BUCKET_ACL
+
+        else:
+            self.acl = acl
+            self.auto_create_acl = auto_create_acl
```

Comment:
I see from the tests that you kept this distinction here to set them to `None`. In this case, I would write it explicitly
```py
            self.acl = None
            self.auto_create_acl = None
```